praktikumsplaner-frontend/frontend/src/components/nachwuchskraefte/NwkCreateDialog.vue

```diff
@@ -0,0 +1,118 @@
+                                    <JahrgangInput
+                                        v-model="nwk"
+                                    ></JahrgangInput>
+                                </v-col>
+                                <v-col cols="6">
+                                    <VorlesungstageSelector
+                                        v-model="nwk"
+                                    ></VorlesungstageSelector>
+                                </v-col>
+                            </v-row>
+                        </v-list-item>
+
+                        <v-row>
+                            <v-list-item>
+                                <StudienrichtungOrAusbildungsrichtungSelect
+                                    v-model="nwk"
+                                ></StudienrichtungOrAusbildungsrichtungSelect>
+                            </v-list-item>
+                        </v-row>
+                    </v-list>
+                    <v-card-actions>
+                        <v-spacer />
+                        <v-btn
+                            color="primary"
+                            outlined
+                            @click="cancel()"
+                        >
+                            Abbrechen
+                        </v-btn>
+                        <v-btn
+                            color="primary"
+                            @click="saveNwk()"
+                        >
+                            Akzeptieren
+                        </v-btn>
+                    </v-card-actions>
+                </v-card>
+            </v-form>
+        </v-dialog>
+        <progress-circular-overlay
+            :loading="loading"
+        ></progress-circular-overlay>
+    </div>
+</template>
+
+<script setup lang="ts">
+import { ref } from "vue";
+import NwkService from "@/api/NwkService";
+import { useRules } from "@/composables/rules";
+import ProgressCircularOverlay from "@/components/common/ProgressCircularOverlay.vue";
+import NwkCreate from "@/types/NwkCreate";
+import VorlesungstageSelector from "@/components/nachwuchskraefte/VorlesungstageSelect.vue";
+import StudienrichtungOrAusbildungsrichtungSelect from "@/components/common/StudienrichtungOrAusbildungsrichtungSelect.vue";
+import NameInput from "@/components/common/NameInput.vue";
+import JahrgangInput from "@/components/common/JahrgangInput.vue";
+import { EventBus } from "@/stores/event-bus";
+
+const visible = ref<boolean>(false);
+const loading = ref<boolean>(false);
+const form = ref<HTMLFormElement>();
+const validationRules = useRules();
+
+const nwk = ref<NwkCreate>(new NwkCreate("", "", "", [], undefined, undefined));
+
+function cancel() {
+    visible.value = false;
+}
+
+function saveNwk() {
+    if (!form.value?.validate()) {
+        return;
+    }
+    loading.value = true;
+    cancel();
+    NwkService.saveNwk(nwk.value).finally(() => {
+        loading.value = false;
+        EventBus.$emit("nwkCreated");
+    });
```

Comment:
the emit should be moved into a `then` bc if an error occurs the event should not be called